### PR TITLE
(feat) Adds optional stash flag to pause item degrade

### PR DESCRIPTION
**Pull request by Senlar**
This adds an optional stash flag that you can add in the stashes list or send with RegisterStash. If set, this flag will cause all item degrading while they are stored in the stash.

Example config included in data\stashes.lua

### CODE REVIEW

File: server.lua [hunk 1]
```diff
@@ -69,7 +69,14 @@ ServerCallback.Register('openInventory', function(source, inv, data)
 					right = Inventory(owner and stash.name..owner or stash.name)
 
 					if not right then
-						right = Inventory.Create(owner and stash.name..owner or stash.name, stash.label or stash.name, inv, stash.slots, 0, stash.weight, owner or false)
+						right = Inventory.Create(owner and stash.name..owner or stash.name, stash.label or stash.name, inv, stash.slots, 0, stash.weight, owner or false, nil, stash.degrade)
+					end
+					if stash.degrade == false then
+						for a,b in pairs(right.items) do
+							if b.metadata.refrigerate then
```
Comment (LukeWasTakenn) [inline server.lua:76]: I feel like checking for `refrigerate` item metadata is maybe a bit too specific naming?

Comment (Senlar) [reply server.lua]: Yup, right. I've changed it to degradeamount

**Review by Senlar — COMMENTED** [review]
Should be all set, unless there's anything else.

**Review by LukeWasTakenn — CHANGES_REQUESTED** [review]
Proper camel case please, `degradeAmount` : (

**Review by LukeWasTakenn — APPROVED** [review]
LGTM, need Linden to go over it as well

**Review by Senlar — COMMENTED** [review]
Corrected new metadata variable to lowerCamelCase

**Comment by thelindat** [comment]
I have some issues with this implementation, not the feature itself; this'll be held until I feel like sitting down to clean it up.

**Comment by Senlar** [comment]
Anything I can help with?

**Pull request closed without merging**